ietf/api/views.py

```diff
@@ -232,6 +233,153 @@ def err(code, text):
+                "reg_type": {"type": "string"},
+                "ticket_type": {"type": "string"},
+                "checkedin": {"type": "boolean"},
+                "is_nomcom_volunteer": {"type": "boolean"},
+                "cancelled": {"type": "boolean"},
+            },
+            "required": ["meeting", "first_name", "last_name", "affiliation", "country_code", "email", "reg_type", "ticket_type", "checkedin", "is_nomcom_volunteer", "cancelled"],
+            "additionalProperties": "false"
+        }
+    }
+)
+
+
+@requires_api_token
+@csrf_exempt
+def api_new_meeting_registration_v2(request):
+    '''REST API to notify the datatracker about new or updated meeting registrations'''
+
+    def _safe_pop(lst):
+        if lst:
+            return lst.pop()
+        else:
+            return None
+
+    def _http_err(code, text):
+        return HttpResponse(text, status=code, content_type="text/plain")
+
+    def _api_response(result):
+        return JsonResponse(data={"result": result})
+
+    if request.method != "POST":
+        return _http_err(405, "Method not allowed")
+
+    if request.content_type != "application/json":
+        return _http_err(415, "Content-Type must be application/json")
+
+    # Validate
+    try:
+        payload = json.loads(request.body)
+        _new_registration_json_validator.validate(payload)
+    except json.decoder.JSONDecodeError as err:
+        return _http_err(400, f"JSON parse error at line {err.lineno} col {err.colno}: {err.msg}")
+    except jsonschema.exceptions.ValidationError as err:
+        return _http_err(400, f"JSON schema error at {err.json_path}: {err.message}")
+    except Exception:
+        return _http_err(400, "Invalid request format")
+
+    # Validate consistency
+    # - if receive multiple records they should be for same meeting, same person (email)
+    if len(payload) > 1:
+        if len(set([r['meeting'] for r in payload])) != 1:
+            return _http_err(400, "Different meeting values")
+        if len(set([r['email'] for r in payload])) != 1:
+            return _http_err(400, "Different email values")
+
+    # Validate meeting
+    number = payload[0]['meeting']
+    try:
+        meeting = Meeting.objects.get(number=number)
+    except Meeting.DoesNotExist:
+        return _http_err(400, "Invalid meeting value: '%s'" % (number, ))
+
+    # Validate email
+    email = payload[0]['email']
+    try:
+        validate_email(email)
+    except ValidationError:
+        return _http_err(400, "Invalid email value: '%s'" % (email, ))
+
+    # handle cancelled. there will be only one record
+    if payload[0]['cancelled']:
+        reg = MeetingRegistration.objects.filter(
+            meeting__number=number,
+            email=email,
+            reg_type=payload[0]['reg_type'],
+            ticket_type=payload[0]['ticket_type']).first()
+        if reg:
+            reg.delete()
+        return HttpResponse('Success', status=202, content_type='text/plain')
```

Comment:
I read this to mean that if you are notifying datatracker of cancelled registrations you will send one request per cancelled record - that is, if someone cancels both hackathon and regular meeting, you'll have to POST two different requests each with one record rather than send both cancellations in the same POST.

If that's correct, it would be good to error if the first payload indicates cancelled and there are more than one records in the payload.

Reply:
Notifications to Datatracker are triggered by the post_save signal on Registration. Regular onsite meeting registration and hackathon registration are each a Registration record, meaning a notification gets sent for each record.

Addressed in commit 4ae730675d89454bd9a0d7b86257f54a661fa808